fix(ivy): ensure that super/sub classes can both update an animation binding in harmony

matsko opened the pull request:
In ivy, because metadata is inherited, this may cause super and sub host bindings on
two components to update an animation binding in parallel. This patch ensures that
the animation transition code can handle this.

Jira-Issue: FW-1220

gkalpak commented on packages/animations/browser/src/render/transition_animation_engine.ts:
```diff
@@ -195,29 +195,21 @@ export class AnimationTransitionNamespace {
 
+    const toState = new StateValue(value, this.id);
     const isObj = value && value.hasOwnProperty('value');
     if (!isObj && fromState) {
```
Won't `fromState` always be truthy now? Is that intended?

matsko commented:
http://test/OCL:244417359:BASE:244417378:1555707481798:30629bbb

mary-poppins commented:
You can preview 195a4a1 at https://pr29978-195a4a1.ngbuilds.io/.
You can preview f997d07 at https://pr29978-f997d07.ngbuilds.io/.
You can preview db4e609 at https://pr29978-db4e609.ngbuilds.io/.
You can preview 4444cc1 at https://pr29978-4444cc1.ngbuilds.io/.
You can preview 327532c at https://pr29978-327532c.ngbuilds.io/.

matsko committed:
fix(ivy): ensure that super/sub classes can both update an animation binding in harmony

In ivy, because metadata is inherited, this may cause super and sub host bindings on
two components to update an animation binding in parallel. This patch ensures that
the animation transition code can handle this.

Jira-Issue: FW-1220

mary-poppins commented:
You can preview 3f6bf6d at https://pr29978-3f6bf6d.ngbuilds.io/.

mary-poppins commented:
You can preview ab6b89a at https://pr29978-ab6b89a.ngbuilds.io/.

gkalpak commented:
Is this superceded by #30022?
